### PR TITLE
Merge snapshot probe into log probe config

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
Add captureContext flag in log probe to capture the context like snapshots

# Motivation
Snapshot probes and log probes are quite a like.

# Additional Notes

### CODE REVIEW

**Review by cimi — APPROVED** [review]
Thanks - looks good!

Left a few minor comments; I skimmed the tests since most of the changes were renames.

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/instrumentation/LogInstrumentor.java [hunk 1]
```diff
@@ -39,33 +38,22 @@
...
+  public LogInstrumentor(
       LogProbe logProbe,
       ClassLoader classLoader,
       ClassNode classNode,
       MethodNode methodNode,
       List<DiagnosticMessage> diagnostics) {
     super(logProbe, classLoader, classNode, methodNode, diagnostics);
-    this.capture = null;
-    this.captureFullState = false;
+    this.capture = logProbe.getCapture();
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/instrumentation/LogInstrumentor.java:55]: yes it's still possible, from deserialization with Moshi
same for sampling.

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java [hunk 2]
```diff
@@ -110,13 +113,119 @@ public void toJson(JsonWriter writer, Segment value) throws IOException {
...
+    @Generated
+    @Override
+    public int hashCode() {
+      return Objects.hash(maxReferenceDepth, maxCollectionSize, maxLength, maxFieldCount);
+    }
+  }
+
+  /** Stores sampling configuration */
+  public static final class Sampling {
+    private final double snapshotsPerSecond;
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java:171]: `Capture` config have independent values that can be added individually:

```
"sampling": {
  "maxReferenceDepth": 2
}
```
so with that, we want to set default values for the others 2.

with sampling we have only one (for now) when we specify the sampling we fill directly the value.

File: dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/agent/LogMessageTemplateSummaryBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  public void undefinedArgTemplate() {
+    LogProbe probe = LogProbe.builder().template("{arg}").build();
+    LogMessageTemplateSummaryBuilder summaryBuilder = new LogMessageTemplateSummaryBuilder(probe);
+    summaryBuilder.addEntry(new Snapshot.CapturedContext());
+    assertEquals("UNDEFINED", summaryBuilder.build());
+  }
+
+  @Test
+  public void argTemplate() {
+    LogProbe probe = LogProbe.builder().template("{arg}").build();
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/agent/LogMessageTemplateSummaryBuilderTest.java:45]: yeah honestly this is not very important. much more to be compliant with the coverage. `template()` is only used for test.

File: dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/agent/DebuggerProductChangesListenerTest.java [hunk 4]
```diff
@@ -167,7 +144,7 @@ public void testMergeConfigWithSingleProbe() {
     DebuggerProductChangesListener listener =
         new DebuggerProductChangesListener(tracerConfig, acceptor);
 
-    SnapshotProbe snapshotProbe = createSnapshotProbe("123");
+    LogProbe snapshotProbe = createSnapshotProbe("123");
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/agent/DebuggerProductChangesListenerTest.java:147]: done

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/LogMessageTemplateSummaryBuilder.java [hunk 5]
```diff
@@ -34,6 +34,9 @@ public void addStack(List<CapturedStackFrame> stack) {}
 
   @Override
   public String build() {
+    if (logProbe.getSegments() == null) {
+      return "";
+    }
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/LogMessageTemplateSummaryBuilder.java:39]: fixed with a more meaningful message

**Pull request merged**